nym-api/src/support/http/mod.rs

```diff
@@ -154,3 +151,50 @@ fn setup_cors() -> Result<Cors> {
 
     Ok(cors)
 }
+
+#[allow(dead_code)]
+// TODO dz Cannot combine `Access-Control-Allow-Credentials: true` with `Access-Control-Allow-Headers: *`
+// https://stackoverflow.com/questions/19743396/cors-cannot-use-wildcard-in-access-control-allow-origin-when-credentials-flag-i
+fn setup_cors() -> CorsLayer {
+    CorsLayer::new()
+        .allow_origin(tower_http::cors::Any)
+        .allow_methods([http::Method::GET, http::Method::POST])
+        .allow_headers(tower_http::cors::Any)
+        .allow_credentials(true)
+}
+
+pub(crate) async fn setup_routes(network_monitor: bool) -> anyhow::Result<Router<AxumAppState>> {
+    // TODO dz serve swagger UI
+
+    let router = Router::new()
+        // https://docs.rs/tower-http/0.1.1/tower_http/trace/index.html
+        // TODO dz use tracing instead of env_logger
+        // https://github.com/tokio-rs/axum/blob/main/examples/tracing-aka-logging/src/main.rs
+        // .layer(
+        //     TraceLayer::new_for_http()
+        //         .make_span_with(DefaultMakeSpan::new().include_headers(true))
+        //         .on_request(DefaultOnRequest::new())
+        //         .on_response(DefaultOnResponse::new().latency_unit(tower_http::LatencyUnit::Micros)),
+        // )
+        .route("/", axum::routing::get(|| async {axum::response::Redirect::permanent("/swagger")}))
+        .route("/swagger", axum::routing::get(hello))
+        .merge(circulating_supply_routes())
+        .merge(nym_contract_cache_routes())
+        .merge(node_status_routes(network_monitor))
+        .merge(nym_network_routes())
+        .merge(status::handlers::api_status_routes())
+        .merge(nym_node_routes())
+        .merge(nym_node_routes_unstable())
+        // CORS layer needs to be "outside" of routes
```

Comment:
what about the elephant in the room also known as ecash? : )

Reply:
ecash routes were added in separate commit